AltStore/Settings/LicensesViewController.swift

```diff
@@ -8,42 +8,81 @@
 
 import UIKit
 
-class LicensesViewController: UIViewController
+class LicensesViewController: UITableViewController
 {
-    private var _didAppear = false
-    
-    @IBOutlet private var textView: UITextView!
-    
-    override func viewWillAppear(_ animated: Bool)
+    private var licenses: [[String: String]] = []
+
+    override func viewDidLoad()
     {
-        super.viewWillAppear(animated)
-        
-        self.view.setNeedsLayout()
-        self.view.layoutIfNeeded()
-        
-        // Fix incorrect initial offset on iPhone SE.
-        self.textView.contentOffset.y = 0
+        super.viewDidLoad()
+        loadLicenses()
     }
-    
-    override func viewDidAppear(_ animated: Bool)
+
+    private func loadLicenses()
     {
-        super.viewDidAppear(animated)
-        
-        _didAppear = true
+        guard let path = Bundle.main.path(forResource: "licenses", ofType: "json") else {
+            dismiss(animated: true)
+            return
+        }
+
+        let url = URL(fileURLWithPath: path)
+
+        guard let data = try? Data(contentsOf: url), let json = try? JSONSerialization.jsonObject(with: data) as? [[String: String]] else {
+            dismiss(animated: true)
+            return
+        }
+
+        licenses = json
+    }
+}
+
+extension LicensesViewController
+{
+    override func numberOfSections(in tableView: UITableView) -> Int
+    {
+        return 1
     }
 
-    override func viewDidLayoutSubviews()
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int
     {
-        super.viewDidLayoutSubviews()
-        
-        self.textView.textContainerInset.left = self.view.layoutMargins.left
-        self.textView.textContainerInset.right = self.view.layoutMargins.right
-        self.textView.textContainer.lineFragmentPadding = 0
-        
-        if !_didAppear
+        return licenses.count
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell
+    {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "licenseListCell", for: indexPath) as! LicenseTableViewCell
+        let license = licenses[indexPath.row]
+
+        switch indexPath.row {
+        case 0:
+            cell.style = .top
+            break
+        case licenses.count - 1:
+            cell.style = .bottom
+            break
+        default:
+            cell.style = .middle
+        }
```

Comment:
Move this switch case comparison to separate function. Like
cell.style = getStyleFor(license: [String: String])